### ISSUE TITLE
scsi.ls fails to run both on CentOS 6 and CentOS 7

### ISSUE THREAD

**Issue by heaje** [saltstack/salt#30565]
The "scsi.ls" module doesn't appear to work correctly on CentOS 7 or 6.  

In CentOS 6, the module fails with this error:
```
[INFO    ] Executing command 'lsscsi -dLsv' in directory '/root'
[ERROR   ] Command 'lsscsi -dLsv' failed with return code: 1
[ERROR   ] output: lsscsi: invalid option -- 's'
Usage: lsscsi   [--classic] [--device] [--generic] [--help] [--hosts]
		[--kname] [--list] [--long] [--protection] [--sysfsroot=PATH]
		[--transport] [--verbose] [--version] [<h:c:t:l>]
  where:
    --classic|-c      alternate output similar to 'cat /proc/scsi/scsi'
    --device|-d       show device node's major + minor numbers
    --generic|-g      show scsi generic device name
    --help|-h         this usage information
    --hosts|-H        lists scsi hosts rather than scsi devices
    --kname|-k        show kernel name instead of device node name
    --list|-L         additional information output one
                      attribute=value per line
    --long|-l         additional information output
    --protection|-p   show data integrity (protection) information
    --sysfsroot=PATH|-y PATH    set sysfs mount point to PATH (def: /sys)
    --transport|-t    transport information for target or, if '--hosts'
                      given, for initiator
    --verbose|-v      output path names where data is found
    --version|-V      output version string and exit
    <h:c:t:l>         filter output list (def: '- - - -' (all))

List SCSI devices or hosts, optionally with additional information
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
IndexError: list index out of range
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 331, in salt_call
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/call.py", line 53, in run
    caller.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/caller.py", line 133, in run
    ret = self.call()
  File "/usr/lib/python2.6/site-packages/salt/cli/caller.py", line 196, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/salt/modules/scsi.py", line 56, in ls_
    ret[key][comps[0]] = comps[1]
IndexError: list index out of range
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 331, in salt_call
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/call.py", line 53, in run
    caller.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/caller.py", line 133, in run
    ret = self.call()
  File "/usr/lib/python2.6/site-packages/salt/cli/caller.py", line 196, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/salt/modules/scsi.py", line 56, in ls_
    ret[key][comps[0]] = comps[1]
IndexError: list index out of range
```

In CentOS 7, it fails with a different error:
```
[INFO    ] Executing command 'lsscsi -dLsv' in directory '/root'
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
ValueError: need more than 1 value to unpack
Traceback (most recent call last):
  File "/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 331, in salt_call
    client.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/call.py", line 53, in run
    caller.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 133, in run
    ret = self.call()
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 196, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/scsi.py", line 36, in ls_
    major, minor = majmin.replace('[', '').replace(']', '').split(':')
ValueError: need more than 1 value to unpack
Traceback (most recent call last):
  File "/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 331, in salt_call
    client.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/call.py", line 53, in run
    caller.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 133, in run
    ret = self.call()
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 196, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/modules/scsi.py", line 36, in ls_
    major, minor = majmin.replace('[', '').replace(']', '').split(':')
ValueError: need more than 1 value to unpack
```

On the CentOS 7 box, here is the output of "lsscsi -dLsv" (the command salt runs to get the SCSI devices):
```
[0:0:0:0]    disk    HP       LOGICAL VOLUME   6.68  /dev/sda [8:0]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x23c7305
  ioerr_cnt=0x1
  iorequest_cnt=0x23c7305
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:0  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:0]
[0:0:0:1]    disk    HP       LOGICAL VOLUME   6.68  /dev/sdb [8:16]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x65e9
  ioerr_cnt=0x1
  iorequest_cnt=0x65e9
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:1  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:1]
[0:0:0:2]    disk    HP       LOGICAL VOLUME   6.68  /dev/sdc [8:32]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x638a
  ioerr_cnt=0x1
  iorequest_cnt=0x638a
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:2  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:2]
[0:0:0:3]    disk    HP       LOGICAL VOLUME   6.68  /dev/sdd [8:48]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x6612
  ioerr_cnt=0x1
  iorequest_cnt=0x6612
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:3  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:3]
[0:0:0:4]    disk    HP       LOGICAL VOLUME   6.68  /dev/sde [8:64]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x64d2
  ioerr_cnt=0x1
  iorequest_cnt=0x64d2
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:4  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:4]
[0:0:0:5]    disk    HP       LOGICAL VOLUME   6.68  /dev/sdf [8:80]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x6547
  ioerr_cnt=0x1
  iorequest_cnt=0x6547
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:5  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:5]
[0:0:0:6]    disk    HP       LOGICAL VOLUME   6.68  /dev/sdg [8:96]  1.20TB
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x65b2
  ioerr_cnt=0x1
  iorequest_cnt=0x65b2
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=30
  type=0
  dir: /sys/bus/scsi/devices/0:0:0:6  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:0:0/0:0:0:6]
[0:3:0:0]    storage HP       P420i            6.68  -               -
  device_blocked=0
  iocounterbits=32
  iodone_cnt=0x5
  ioerr_cnt=0x0
  iorequest_cnt=0x5
  queue_depth=1020
  queue_type=none
  scsi_level=6
  state=running
  timeout=0
  type=12
  dir: /sys/bus/scsi/devices/0:3:0:0  [/sys/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:3:0/0:3:0:0]
```

```
Salt Version:
           Salt: 2015.8.3
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.5.0
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: 0.3.7
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: centos 7.1.1503 Core
        machine: x86_64
        release: 3.10.0-229.el7.x86_64
         system: CentOS Linux 7.1.1503 Core
```

**Comment by ssgward** [comment]
@heaje - thanks for the detailed bug report.

**Comment by sjmh** [comment]
I have a PR ready for this, but I'm wondering what the opinion would be on what to do if '-s' isn't available ( as is the case with <= RHEL6 distributions )

Should it simply exit with an error or should it try to fallback to using lsscsi without the -s?

My opinion would be to simply return an error, rather than give them back the data without the size value.  Seems like if it were to fallback, it may cause issues further down the line if they're expecting the size value to be there in a state.

**Comment by heaje** [comment]
Based on conversation in #salt, it would make sense to create a paramter called something like get_size.  If true, use -s.  If running on a centos 6 box with get_size=true, we should catch the exception with a sane error message.

**Comment by rallytime** [comment]
@heaje Are you in a position to give #30659 a try and let us know if this is resolved for you?

**Comment by heaje** [comment]
@rallytime ,

I just tested this out on both a CentOS 6 and 7 box.  It works just fine on both.  Just a thought, but perhaps instead of simply returning "lsscsi: invalid option -- 's'" on the CentOS 6 box when get_size=true, it would be better to return perhaps that error with a suggestion to try running with get_size=false.

**Comment by rallytime** [comment]
@heaje Great suggestion, and thanks for testing!

@sjmh Want to edit your PR according to the suggestion above?

**Comment by rallytime** [comment]
Fixed via #30659
